### PR TITLE
feat: add multiple rewards model support

### PR DESCRIPTION
**Pull request by wuxibin89**

### CODE REVIEW

**Review by catqaq — COMMENTED** [review]
now, save checkpoint is not supported in ppo_ray. so, save_steps must be -1.

**Review by catqaq — COMMENTED** [review]
need shell script?

**Comment by hijkzzz** [comment]
I can merge this request first and fix the problem with a new MR

**Comment by wuxibin89** [comment]
> now, save checkpoint is not supported in ppo_ray. so, save_steps must be -1.

train_ppo_ray.py reuse `openrlhf.trainer.ppo_trainer.py`, so set save_steps will enable checkpoint properly. Need some test anyway.

**Comment by wuxibin89** [comment]
> After a quick look at the code, I can't find any problems for the time being. However, I currently don’t have the time or resources to run an actual verification. Please help test it when you all are free and report on the performance.
> 
> need shell script? train_ppo_llama_multi_reward_ray.sh？

I have test multiple rewards ppo with following script. The training process works fine, however the convergence needs further verification.

```bash
-ray job submit --address="http://127.0.0.1:8265" \
    --runtime-env-json='{"working_dir": "/openrlhf", "pip": "/openrlhf/requirements.txt"}' \
    --no-wait \
    -- python3 examples/train_ppo_ray.py \
    --ref_num_nodes 1 \
    --ref_num_gpus_per_node 1 \
    --reward_num_nodes 1 \
    --reward_num_gpus_per_node 1 \
    --critic_num_nodes 1 \
    --critic_num_gpus_per_node 2 \
    --actor_num_nodes 1 \
    --actor_num_gpus_per_node 2 \
    --pretrain meta-llama/Llama-2-7b-hf \
    --critic_pretrain meta-llama/Llama-2-7b-hf,meta-llama/Llama-2-7b-hf \
    --save_path /openrlhf/examples/test_scripts/ckpt/7b_llama \
    --micro_train_batch_size 8 \
    --train_batch_size 128 \
    --micro_rollout_batch_size 16 \
    --rollout_batch_size 1024 \
    --max_epochs 1 \
    --prompt_max_len 1024 \
    --generate_max_len 1024 \
    --zero_stage 2 \
    --bf16 \
    --actor_learning_rate 5e-7 \
    --critic_learning_rate 9e-6 \
    --inference_tp_size 1 \
    --init_kl_coef 0.01 \
    --prompt_data Open-Orca/OpenOrca,Dahoas/full-hh-rlhf,tasksource/oasst1_pairwise_rlhf_reward \
    --prompt_data_probs 0.4,0.5,0.1 \
    --max_samples 80000 \
    --normalize_reward \
    --actor_init_on_gpu \
    --adam_offload \
    --gradient_checkpointing
```

**Comment by cattleyawlotus** [comment]
After looking at the code, is multiple reward models not yet supported for ppo without ray?
There is no process on reward model path list in ppo training.
